### PR TITLE
OCPCRT-275: Require `jira/valid-reference` label for all PR merges

### PR DESCRIPTION
**Pull request by bradmwilliams**
This PR adds the `jira/valid-reference`, as a required label for all repos that provide content into a release.  The list of repos was generated via the following command:
```
$ oc adm release info --commit-urls registry.ci.openshift.org/ocp/release:4.14.0-0.nightly-2023-10-22-172722 -o json | jq -r '.references.spec.tags[].annotations["io.openshift.build.source-location"]'| grep github | cut -d'/' -f 4- | sort -u 
openshift/agent-installer-utils
openshift/alibaba-cloud-csi-driver
openshift/alibaba-disk-csi-driver-operator
openshift/apiserver-network-proxy
openshift/assisted-installer
openshift/assisted-installer-agent
openshift/assisted-service
openshift/aws-ebs-csi-driver
openshift/aws-ebs-csi-driver-operator
openshift/aws-pod-identity-webhook
openshift/azure-disk-csi-driver
openshift/azure-disk-csi-driver-operator
openshift/azure-file-csi-driver
openshift/azure-file-csi-driver-operator
openshift/azure-workload-identity
openshift/baremetal-operator
openshift/baremetal-runtimecfg
openshift/bond-cni
openshift/builder
openshift/cloud-credential-operator
openshift/cloud-network-config-controller
openshift/cloud-provider-alibaba-cloud
openshift/cloud-provider-aws
openshift/cloud-provider-azure
openshift/cloud-provider-gcp
openshift/cloud-provider-ibm
openshift/cloud-provider-kubevirt
openshift/cloud-provider-nutanix
openshift/cloud-provider-openstack
openshift/cloud-provider-powervs
openshift/cloud-provider-vsphere
openshift/cluster-api
openshift/cluster-api-operator
openshift/cluster-api-provider-alibaba
openshift/cluster-api-provider-aws
openshift/cluster-api-provider-azure
openshift/cluster-api-provider-baremetal
openshift/cluster-api-provider-gcp
openshift/cluster-api-provider-ibmcloud
openshift/cluster-api-provider-libvirt
openshift/cluster-api-provider-ovirt
openshift/cluster-api-provider-vsphere
openshift/cluster-authentication-operator
openshift/cluster-autoscaler-operator
openshift/cluster-baremetal-operator
openshift/cluster-bootstrap
openshift/cluster-capi-operator
openshift/cluster-cloud-controller-manager-operator
openshift/cluster-config-operator
openshift/cluster-control-plane-machine-set-operator
openshift/cluster-csi-snapshot-controller-operator
openshift/cluster-dns-operator
openshift/cluster-etcd-operator
openshift/cluster-image-registry-operator
openshift/cluster-ingress-operator
openshift/cluster-kube-apiserver-operator
openshift/cluster-kube-controller-manager-operator
openshift/cluster-kube-scheduler-operator
openshift/cluster-kube-storage-version-migrator-operator
openshift/cluster-machine-approver
openshift/cluster-monitoring-operator
openshift/cluster-network-operator
openshift/cluster-node-tuning-operator
openshift/cluster-olm-operator
openshift/cluster-openshift-apiserver-operator
openshift/cluster-openshift-controller-manager-operator
openshift/cluster-policy-controller
openshift/cluster-samples-operator
openshift/cluster-storage-operator
openshift/cluster-update-keys
openshift/cluster-version-operator
openshift/configmap-reload
openshift/console
openshift/console-operator
openshift/containernetworking-plugins
openshift/coredns
openshift/csi-driver-manila-operator
openshift/csi-driver-nfs
openshift/csi-driver-shared-resource
openshift/csi-driver-shared-resource-operator
openshift/csi-external-attacher
openshift/csi-external-provisioner
openshift/csi-external-resizer
openshift/csi-external-snapshotter
openshift/csi-livenessprobe
openshift/csi-node-driver-registrar
openshift/driver-toolkit
openshift/egress-router-cni
openshift/etcd
openshift/gcp-pd-csi-driver
openshift/gcp-pd-csi-driver-operator
openshift/hypershift
openshift/ibm-powervs-block-csi-driver
openshift/ibm-powervs-block-csi-driver-operator
openshift/ibm-vpc-block-csi-driver
openshift/ibm-vpc-block-csi-driver-operator
openshift/ibm-vpc-node-label-updater
openshift/image-customization-controller
openshift/image-registry
openshift/images
openshift/insights-operator
openshift/installer
openshift/ironic-agent-image
openshift/ironic-image
openshift/ironic-rhcos-downloader
openshift/ironic-static-ip-manager
openshift/k8s-prometheus-adapter
openshift/kube-rbac-proxy
openshift/kubernetes
openshift/kubernetes-autoscaler
openshift/kubernetes-kube-storage-version-migrator
openshift/kube-state-metrics
openshift/kubevirt-csi-driver
openshift/kuryr-kubernetes
openshift/machine-api-operator
openshift/machine-api-provider-aws
openshift/machine-api-provider-azure
openshift/machine-api-provider-gcp
openshift/machine-api-provider-ibmcloud
openshift/machine-api-provider-nutanix
openshift/machine-api-provider-openstack
openshift/machine-api-provider-powervs
openshift/machine-config-operator
openshift/machine-os-images
openshift/monitoring-plugin
openshift/multus-admission-controller
openshift/multus-cni
openshift/multus-networkpolicy
openshift/must-gather
openshift/network-metrics-daemon
openshift/network-tools
openshift/node_exporter
openshift/oauth-apiserver
openshift/oauth-proxy
openshift/oauth-server
openshift/oc
openshift/oc-mirror
openshift/openshift-apiserver
openshift/openshift-controller-manager
openshift/openshift-state-metrics
openshift/openstack-cinder-csi-driver-operator
openshift/operator-framework-catalogd
openshift/operator-framework-olm
openshift/operator-framework-operator-controller
openshift/operator-framework-rukpak
openshift/origin
openshift/ovirt-csi-driver
openshift/ovirt-csi-driver-operator
openshift/ovn-kubernetes
openshift/platform-operators
openshift/prometheus
openshift/prometheus-alertmanager
openshift/prometheus-operator
openshift/prom-label-proxy
openshift/route-controller-manager
openshift/route-override-cni
openshift/router
openshift/sdn
openshift/service-ca-operator
openshift/telemeter
openshift/thanos
openshift/vmware-vsphere-csi-driver
openshift/vmware-vsphere-csi-driver-operator
openshift/vsphere-problem-detector
openshift/whereabouts-cni
operator-framework/operator-marketplace
```

### CODE REVIEW

**Comment by bparees** [comment]
let's be sure to hold this until 4.14 GA is done (so we don't add confusion/disruption to that work) and also let's aim to merge it at the start of the US day so we can be around to help anyone who's surprised/confused (though communication has been sent about this coming)

**Comment by bradmwilliams** [comment]
/hold
Waiting for post 4.14 GA

**Comment by bparees** [comment]
@bradmwilliams  looks like it needs a rebase but otherwise i think we are ready to proceed here

**Comment by bradmwilliams** [comment]
@bparees My third rebase of the the day!  Fingers crossed no more breaking changes before we can get this guy merged! :crossed_fingers:

File: core-services/prow/02_config/openshift-agent-team/fleeting/_prowconfig.yaml [hunk 1]
```diff
@@ -31,17 +31,3 @@ tide:
...
-    - lgtm
-    missingLabels:
-    - backports/unvalidated-commits
-    - do-not-merge/hold
-    - do-not-merge/invalid-owners-file
-    - do-not-merge/work-in-progress
-    - jira/invalid-bug
-    - needs-rebase
-    repos:
-    - openshift-agent-team/fleeting
```
Comment (bparees) [inline core-services/prow/02_config/openshift-agent-team/fleeting/_prowconfig.yaml:47]: any idea why this is being touched?

Comment (bparees) [reply core-services/prow/02_config/openshift-agent-team/fleeting/_prowconfig.yaml:47]: ah, this is removing a duplicate entry

Comment (bradmwilliams) [reply core-services/prow/02_config/openshift-agent-team/fleeting/_prowconfig.yaml:47]: There were 2 files that contained duplicate entries:

1. core-services/prow/02_config/openshift-agent-team/fleeting/_prowconfig.yaml
2. core-services/prow/02_config/openshift/operator-framework-olm/_prowconfig.yaml

I verified they were dups and removed them.

**Pull request merged**